app/src/main/cpp/src/3rd_party/android-helpers/openssl/build_openssl.sh

```diff
@@ -0,0 +1,99 @@
+patch -p1 -N <<EOP
+--- old/Configurations/unix-Makefile.tmpl   2018-09-11 14:48:19.000000000 +0200
++++ new/Configurations/unix-Makefile.tmpl   2018-10-18 09:06:27.282007245 +0200
+@@ -43,12 +43,17 @@
+      # will return the name from shlib(\$libname) with any SO version number
+      # removed.  On some systems, they may therefore return the exact same
+      # string.
+-     sub shlib {
++     sub shlib_simple {
+          my \$lib = shift;
+          return () if \$disabled{shared} || \$lib =~ /\\.a$/;
+-         return \$unified_info{sharednames}->{\$lib}. \$shlibvariant. '\$(SHLIB_EXT)';
++
++         if (windowsdll()) {
++             return \$lib . '\$(SHLIB_EXT_IMPORT)';
++         }
++         return \$lib .  '\$(SHLIB_EXT_SIMPLE)';
+      }
+-     sub shlib_simple {
++     
++   sub shlib {
+          my \$lib = shift;
+          return () if \$disabled{shared} || \$lib =~ /\\.a$/;
+
+EOP
+
+##### export ndk directory. Required by openssl-build-scripts #####
+export ANDROID_NDK
+
+##### build-function #####
+build_the_thing() {
+    TOOLCHAIN=$ANDROID_NDK/toolchains/llvm/prebuilt/linux-x86_64
+    export PATH=$TOOLCHAIN/$TRIBLE/bin:$TOOLCHAIN/bin:"$PATH"
+    ./Configure $SSL_TARGET $OPTIONS -fuse-ld="$TOOLCHAIN/$TRIBLE/bin/ld" && \
+    make -j4 && \
+    make install DESTDIR=$DESTDIR || exit 128
+}
+
+##### set variables according to build-tagret #####
+case $4 in
+armeabi)
+    TRIBLE="arm-linux-androideabi"
+    TC_NAME="arm-linux-androideabi-4.9"
+    #OPTIONS="--target=armv5te-linux-androideabi -mthumb -fPIC -latomic -D__ANDROID_API__=$API_LEVEL"
+    OPTIONS="--target=armv5te-linux-androideabi -mthumb -fPIC -latomic -D__ANDROID_API__=$API_LEVEL"
+    DESTDIR="$BUILD_DIR/armeabi"
+    SSL_TARGET="android-arm"
+    build_the_thing
+;;
+armeabi-v7a)
+    TRIBLE="arm-linux-androideabi"
+    TC_NAME="arm-linux-androideabi-4.9"
+    OPTIONS="--target=armv7a-linux-androideabi -Wl,--fix-cortex-a8 -fPIC -D__ANDROID_API__=$API_LEVEL"
+    DESTDIR="$BUILD_DIR/armeabi-v7a"
+    SSL_TARGET="android-arm"
+    build_the_thing
+;;
+x86)
+    TRIBLE="i686-linux-android"
+    TC_NAME="x86-4.9"
+    OPTIONS="-fPIC -D__ANDROID_API__=${API_LEVEL}"
+    DESTDIR="$BUILD_DIR/x86"
+    SSL_TARGET="android-x86"
+    build_the_thing
+;;
+x86_64)
+    TRIBLE="x86_64-linux-android"
+    TC_NAME="x86_64-4.9"
+    OPTIONS="-fPIC -D__ANDROID_API__=${API_LEVEL}"
+    DESTDIR="$BUILD_DIR/x86_64"
+    SSL_TARGET="android-x86_64"
+    build_the_thing
+;;
+arm64-v8a)
+    TRIBLE="aarch64-linux-android"
+    TC_NAME="aarch64-linux-android-4.9"
+    OPTIONS="-fPIC -D__ANDROID_API__=${API_LEVEL}"
+    DESTDIR="$BUILD_DIR/arm64-v8a"
+    SSL_TARGET="android-arm64"
+    build_the_thing
+;;
+esac
+
+#### copy libraries and includes to output-directory #####
+mkdir -p $OUT_DIR/include/openssl
+cp -R $DESTDIR/usr/local/include/* $OUT_DIR/include
+#mkdir -p $OUT_DIR/lib/openssl
+cp -R $DESTDIR/usr/local/lib/*.so $OUT_DIR/lib
+
+echo Success
```

Comment:
@VadymLuchko add empty line

Reply:
done in 048ae59c19ca4e61f504361cf27bd9190b2e5830